Inconsistent/random .Segments.Path.Writable result with v24+

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### What happened?

After upgrading from v23.20.3 → v24.0.2, and currently on v24.0.4, I've noticed a `.Segments.Path.Writable` template condition:

```json
        {
          "style": "diamond",
          "leading_diamond": "\ue0b6",
          "trailing_diamond": "\ue0b0",
          "template": "{{ if .WSL }}WSL at {{ end }}{{.Icon}} ",
          "foreground": "black",
          "background": "magenta",
          "type": "os",
          "background_templates": [
            "{{ if .Root }}red{{ end }}",
            "{{ if not .Segments.Path.Writable }}white{{ end }}"
          ]
        },
```

stopped working.  More accurately it fails to indicate a writable directory more often than it works as expected (white indicates non-writable directory, while purple is normal/writable):

Cmd | Bash
---|---
![cmd](https://github.com/user-attachments/assets/2de49002-5b94-46d9-aa5a-34dbb890eeeb) | ![bash](https://github.com/user-attachments/assets/95045a43-1366-493e-8581-07de33d7297d)



### Theme

<details>
<summary>Custom</summary>

```jsonc
{
  "$schema": "https://raw.githubusercontent.com/JanDeDobbeleer/oh-my-posh/main/themes/schema.json",
  "blocks": [
    {
      "type": "prompt",
      "alignment": "right",
      "segments": [
        {
          "properties": {
            "style": "round"
          },
          "style": "plain",
          "template": "{{ .FormattedMs }} | ",
          "foreground": "darkGray",
          "background": "",
          "type": "executiontime"
        },
        {
          "style": "plain",
          "template": "{{ if .SSHSession }}\uf817 {{ end }}{{ .UserName }}@{{ .HostName }}",
          "foreground": "green",
          "background": "",
          "type": "session"
        },
        {
          "properties": {
            "root_icon": "root"
          },
          "style": "plain",
          "template": " as \uf0e7",
          "foreground": "red",
          "background": "",
          "type": "root"
        }
      ]
    },
    {
      "type": "prompt",
      "alignment": "left",
      "segments": [
        {
          "style": "diamond",
          "leading_diamond": "\ue0b6",
          "trailing_diamond": "\ue0b0",
          "template": "{{ if .WSL }}WSL at {{ end }}{{.Icon}} ",
          "foreground": "black",
          "background": "magenta",
          "type": "os",
          "background_templates": [
            "{{ if .Root }}red{{ end }}",
            "{{ if not .Segments.Path.Writable }}white{{ end }}"
          ]
        },
        {
          "properties": {
            "folder_separator_icon": " \ue0b1 ",
            "home_icon": "\udb80\udedc ",
            "max_depth": 4,
            "style": "agnoster_short"
          },
          "style": "diamond",
          "leading_diamond": "<black,yellow>\ue0b0</>",
          "trailing_diamond": "\ue0b0",
          "template": " {{ path .Path .Location }} ",
          "foreground": "black",
          "powerline_symbol": "<yellow,black>\ue0b0",
          "background": "yellow",
          "type": "path",
          "alias": "Path"
        },
        {
          "properties": {
            "commit_icon": "\uf417 ",
            "fetch_status": true,
            "fetch_upstream_icon": true,
            "fetch_user": true,
            "tag_icon": "\uf412 ",
            "untracked_modes": {
              "*": "no"
            }
          },
          "style": "diamond",
          "leading_diamond": "<transparent,cyan>\ue0b0</>",
          "trailing_diamond": "\ue0b4",
          "template": " {{ .UpstreamIcon }} {{ .HEAD }} {{ .BranchStatus }}{{ if .Working.Changed }} \uf044 {{ .Working.String }}{{ end }}{{ if and (.Staging.Changed) (.Working.Changed) }} |{{ end }}{{ if .Staging.Changed }} \uf046 {{ .Staging.String }} {{ .User.Email }}{{ end }} ",
          "foreground": "black",
          "background": "cyan",
          "type": "git"
        }
      ],
      "newline": true
    },
    {
      "type": "prompt",
      "alignment": "left",
      "segments": [
        {
          "style": "plain",
          "template": "{{ if eq .Shell \"cmd\" }}❯{{ else if .Root }}#{{ else }}${{ end }} ",
          "foreground": "lightWhite",
          "background": "",
          "type": "text"
        }
      ],
      "newline": true
    }
  ],
  "version": 3,
  "auto_upgrade": true
}
```
</details>

### What OS are you seeing the problem on?

Windows

### Which shell are you using?

bash, cmd

### Log output

```Shell
Version: 24.0.4

Shell: cmd (clink v1.7.4.d1920c)

Prompt:

                                                                                                                                    stanio@staniomob
  󰋜  
❯

Segments:

ConsoleTitle(false)                        -   0 ms
Executiontime(false)                       -   0 ms
Session(true)                              -   1 ms
Root(false)                                -   0 ms
Os(true)                                   -   0 ms
Path(true)                                 -   3 ms
Git(false)                                 -  59 ms
Text(true)                                 -   0 ms

Run duration: 67.1103ms

Cache path: C:\Users\stanio\AppData\Local\oh-my-posh

Config path: C:\Users\stanio\AppData\Local\Programs\oh-my-posh\themes\stanio.omp.jsonc

Logs:

[DEBUG] 08:14:02.068 debug.go:50 → debug mode enabled
[DEBUG] 08:14:02.070 debug.go:50 → plain mode enabled
[DEBUG] 08:14:02.070 terminal.go:Getenv:161 → C:\Users\stanio\AppData\Local
[TRACE] 08:14:02.070 terminal.go:Getenv(LOCALAPPDATA) - 570.4µs
[TRACE] 08:14:02.070 terminal.go:CachePath() - 570.4µs
[DEBUG] 08:14:02.070 terminal.go:65 → loading cache file: C:\Users\stanio\AppData\Local\oh-my-posh\omp.cache
[DEBUG] 08:14:02.071 terminal.go:65 → loading cache key: upgrade_check
[TRACE] 08:14:02.071 terminal.go(C:\Users\stanio\AppData\Local\oh-my-posh\omp.cache) - 535.3µs
[DEBUG] 08:14:02.071 terminal.go:Getenv:161 → C:\Users\stanio\AppData\Local
[TRACE] 08:14:02.071 terminal.go:Getenv(LOCALAPPDATA) - 0s
[TRACE] 08:14:02.071 terminal.go:CachePath() - 575.2µs
[DEBUG] 08:14:02.071 terminal.go:65 → loading cache file: C:\Users\stanio\AppData\Local\oh-my-posh\omp.cache.7ad8ac9c-0aed-45f5-a9a4-c55ea590774a
[TRACE] 08:14:02.071 terminal.go(C:\Users\stanio\AppData\Local\oh-my-posh\omp.cache.7ad8ac9c-0aed-45f5-a9a4-c55ea590774a) - 0s
[TRACE] 08:14:02.071 terminal.go:setPromptCount() - 0s
[TRACE] 08:14:02.071 terminal.go:GOOS() - 0s
[DEBUG] 08:14:02.072 terminal.go:setPwd:194 → C:\Users\stanio
[TRACE] 08:14:02.072 terminal.go:setPwd() - 564.2µs
[DEBUG] 08:14:02.072 terminal.go:Getenv:161 → C:\Users\stanio\AppData\Local\Programs\oh-my-posh\themes\stanio.omp.jsonc
[TRACE] 08:14:02.072 terminal.go:Getenv(POSH_THEME) - 0s
[DEBUG] 08:14:02.072 terminal.go:ResolveConfigPath:93 → config set using POSH_THEME: C:\Users\stanio\AppData\Local\Programs\oh-my-posh\themes\stanio.omp.jsonc
[TRACE] 08:14:02.072 terminal.go:ResolveConfigPath() - 0s
[TRACE] 08:14:02.072 terminal.go:Init() - 3.5111ms
[TRACE] 08:14:02.072 terminal.go:Shell() - 0s
[TRACE] 08:14:02.072 terminal.go:Flags() - 0s
[TRACE] 08:14:02.073 load.go:loadConfig() - 1.1131ms
[TRACE] 08:14:02.073 terminal.go:Flags() - 0s
[DEBUG] 08:14:02.073 debug.go:60 → terminal program: Windows Terminal
[DEBUG] 08:14:02.073 debug.go:60 → terminal shell: shell
[DEBUG] 08:14:02.073 terminal.go:Getenv:161 → NO DATA
[TRACE] 08:14:02.073 terminal.go:Getenv(OMP_CACHE_DISABLED) - 0s
[TRACE] 08:14:02.074 terminal_windows.go:WindowsRegistryKeyValue(HKEY_CURRENT_USER\Software\Microsoft\Windows\DWM\ColorizationColor) - 0s
[TRACE] 08:14:02.074 terminal.go:GOOS() - 0s
[TRACE] 08:14:02.074 terminal.go:GOOS() - 0s
[TRACE] 08:14:02.074 terminal.go:GOOS() - 0s
[TRACE] 08:14:02.074 terminal.go:GOOS() - 0s
[TRACE] 08:14:02.074 terminal.go:GOOS() - 0s
[TRACE] 08:14:02.074 terminal.go:GOOS() - 0s
[TRACE] 08:14:02.074 terminal.go:GOOS() - 0s
[TRACE] 08:14:02.074 terminal.go:GOOS() - 0s
[TRACE] 08:14:02.074 terminal.go:GOOS() - 0s
[TRACE] 08:14:02.074 terminal.go:GOOS() - 0s
[TRACE] 08:14:02.074 terminal.go:GOOS() - 0s
[TRACE] 08:14:02.074 terminal.go:GOOS() - 0s
[TRACE] 08:14:02.074 terminal.go:GOOS() - 0s
[TRACE] 08:14:02.074 terminal.go:GOOS() - 0s
[TRACE] 08:14:02.074 terminal.go:GOOS() - 0s
[TRACE] 08:14:02.074 terminal.go:GOOS() - 0s
[TRACE] 08:14:02.074 terminal.go:GOOS() - 0s
[TRACE] 08:14:02.074 terminal.go:GOOS() - 0s
[TRACE] 08:14:02.074 terminal.go:GOOS() - 0s
[DEBUG] 08:14:02.074 terminal_windows.go:WindowsRegistryKeyValue:213 → ColorizationColor(DWORD): 0xC450506C
[TRACE] 08:14:02.074 colors_windows.go:GetAccentColor() - 584.4µs
[TRACE] 08:14:02.074 colors.go:SetAccentColor() - 584.4µs
[TRACE] 08:14:02.074 terminal.go:Shell() - 0s
[DEBUG] 08:14:02.074 terminal.go:Getenv:161 → clink v1.7.4.d1920c
[TRACE] 08:14:02.074 terminal.go:Getenv(POSH_SHELL_VERSION) - 0s
[DEBUG] 08:14:02.074 debug.go:PrintDebug:23 → segment: Title
[TRACE] 08:14:02.074 text.go:Render() - 0s
[TRACE] 08:14:02.074 terminal.go:Flags() - 0s
[DEBUG] 08:14:02.074 terminal.go:Getenv:161 → 8
[TRACE] 08:14:02.074 terminal.go:Getenv(POSH_CURSOR_LINE) - 0s
[DEBUG] 08:14:02.074 terminal.go:Getenv:161 → NO DATA
[TRACE] 08:14:02.074 terminal.go:Getenv(POSH_CURSOR_COLUMN) - 0s
[TRACE] 08:14:02.074 terminal.go:Flags() - 0s
[TRACE] 08:14:02.074 terminal.go:Flags() - 0s
[TRACE] 08:14:02.074 terminal.go:Flags() - 0s
[DEBUG] 08:14:02.074 terminal_windows.go:56 → C:\Users\stanio
[TRACE] 08:14:02.074 terminal.go:GOOS() - 0s
[TRACE] 08:14:02.074 terminal.go:Flags() - 0s
[DEBUG] 08:14:02.074 segment.go:Execute:103 → segment: Root
[TRACE] 08:14:02.074 terminal.go:Flags() - 0s
[DEBUG] 08:14:02.074 terminal_windows.go:56 → C:\Users\stanio
[TRACE] 08:14:02.074 terminal.go:GOOS() - 0s
[DEBUG] 08:14:02.074 segment.go:Execute:103 → segment: Executiontime
[DEBUG] 08:14:02.074 terminal_windows.go:56 → C:\Users\stanio
[TRACE] 08:14:02.074 terminal.go:GOOS() - 0s
[DEBUG] 08:14:02.074 segment.go:Execute:103 → segment: Session
[TRACE] 08:14:02.074 terminal_windows.go:Root() - 0s
[DEBUG] 08:14:02.074 executiontime.go:Enabled:55 → always_enabled: false
[TRACE] 08:14:02.074 terminal.go:ExecutionTime() - 0s
[DEBUG] 08:14:02.074 terminal.go:Getenv:161 → NO DATA
[TRACE] 08:14:02.074 terminal.go:Getenv(SSH_CONNECTION) - 0s
[DEBUG] 08:14:02.074 executiontime.go:Enabled:57 → threshold: 500.000000
[TRACE] 08:14:02.074 text.go:Render(plain) - 0s
[DEBUG] 08:14:02.074 terminal.go:Getenv:161 → NO DATA
[TRACE] 08:14:02.074 terminal.go:Getenv(SSH_CLIENT) - 0s
[TRACE] 08:14:02.074 terminal_windows.go:Root() - 0s
[TRACE] 08:14:02.075 terminal.go:Shell() - 0s
[TRACE] 08:14:02.075 terminal.go:StatusCodes() - 0s
[TRACE] 08:14:02.075 terminal_windows.go:IsWsl() - 0s
[DEBUG] 08:14:02.075 terminal_windows.go:56 → C:\Users\stanio
[TRACE] 08:14:02.075 terminal_windows.go:IsWsl() - 0s
[TRACE] 08:14:02.075 terminal.go:GOOS() - 0s
[TRACE] 08:14:02.075 terminal.go:GOOS() - 0s
[TRACE] 08:14:02.075 terminal.go:GOOS() - 0s
[TRACE] 08:14:02.075 terminal.go:GOOS() - 0s
[TRACE] 08:14:02.075 terminal.go:GOOS() - 0s
[TRACE] 08:14:02.075 terminal.go:GOOS() - 0s
[TRACE] 08:14:02.075 terminal.go:GOOS() - 0s
[TRACE] 08:14:02.075 terminal.go:GOOS() - 0s
[TRACE] 08:14:02.075 terminal.go:GOOS() - 0s
[DEBUG] 08:14:02.075 terminal.go:User:331 → stanio
[TRACE] 08:14:02.075 terminal.go:User() - 0s
[DEBUG] 08:14:02.075 terminal.go:Host:348 → staniomob
[TRACE] 08:14:02.075 terminal.go:Host() - 0s
[TRACE] 08:14:02.075 terminal.go:GOOS() - 0s
[DEBUG] 08:14:02.075 terminal.go:Getenv:161 → NO DATA
[TRACE] 08:14:02.075 terminal.go:Getenv(SHLVL) - 0s
[TRACE] 08:14:02.075 terminal.go:TemplateCache() - 1.0762ms
[TRACE] 08:14:02.075 terminal.go:TemplateCache() - 0s
[TRACE] 08:14:02.075 text.go:Render({{ if .SSHSession }} {{ end }}{{ .UserName }}@{{ .HostName }}) - 0s
[TRACE] 08:14:02.075 text.go:Render(plain) - 0s
[TRACE] 08:14:02.075 text.go:Render(plain) - 0s
[DEBUG] 08:14:02.076 terminal_windows.go:TerminalWidth:114 → terminal width: 148
[TRACE] 08:14:02.076 terminal_windows.go:TerminalWidth() - 591.2µs
[TRACE] 08:14:02.076 terminal.go:Shell() - 0s
[TRACE] 08:14:02.076 terminal.go:Shell() - 0s
[TRACE] 08:14:02.076 terminal.go:Flags() - 0s
[DEBUG] 08:14:02.076 terminal_windows.go:56 → C:\Users\stanio
[TRACE] 08:14:02.076 terminal.go:GOOS() - 0s
[DEBUG] 08:14:02.076 segment.go:Execute:103 → segment: Git
[TRACE] 08:14:02.076 terminal.go:Flags() - 0s
[TRACE] 08:14:02.076 terminal.go:Flags() - 0s
[TRACE] 08:14:02.076 terminal.go:GOOS() - 0s
[DEBUG] 08:14:02.076 terminal_windows.go:56 → C:\Users\stanio
[DEBUG] 08:14:02.076 terminal_windows.go:56 → C:\Users\stanio
[TRACE] 08:14:02.076 terminal.go:GOOS() - 0s
[DEBUG] 08:14:02.076 segment.go:Execute:103 → segment: Path
[TRACE] 08:14:02.076 terminal.go:GOOS() - 0s
[DEBUG] 08:14:02.076 segment.go:Execute:103 → segment: Os
[TRACE] 08:14:02.076 terminal.go:GOOS() - 0s
[DEBUG] 08:14:02.076 path.go:145 → display_cygpath: false
[TRACE] 08:14:02.076 terminal.go:GOOS() - 0s
[TRACE] 08:14:02.076 terminal.go:GOOS() - 0s
[DEBUG] 08:14:02.076 os.go:Enabled:33 → 
[TRACE] 08:14:02.076 terminal.go:PathSeparator() - 0s
[TRACE] 08:14:02.076 terminal.go:TemplateCache() - 0s
[TRACE] 08:14:02.076 terminal.go:Shell() - 0s
[TRACE] 08:14:02.076 terminal.go:Shell() - 0s
[TRACE] 08:14:02.076 terminal.go:GOOS() - 0s
[DEBUG] 08:14:02.077 path.go:setMappedLocations:552 → mapped_locations_enabled: true
[DEBUG] 08:14:02.077 path.go:setMappedLocations:553 → 
[DEBUG] 08:14:02.077 path.go:setMappedLocations:554 → 
[DEBUG] 08:14:02.077 terminal_windows.go:56 → C:\Users\stanio
[TRACE] 08:14:02.077 terminal.go:GOOS() - 0s
[TRACE] 08:14:02.077 terminal.go:PathSeparator() - 0s
[TRACE] 08:14:02.077 terminal.go:GOOS() - 0s
[TRACE] 08:14:02.077 terminal.go:GOOS() - 0s
[TRACE] 08:14:02.077 terminal.go:GOOS() - 0s
[DEBUG] 08:14:02.077 path.go:setMappedLocations:555 → 󰋜
[DEBUG] 08:14:02.078 path.go:setMappedLocations:560 → mapped_locations: map[]
[TRACE] 08:14:02.078 terminal.go:GOOS() - 0s
[TRACE] 08:14:02.078 terminal.go:PathSeparator() - 0s
[TRACE] 08:14:02.078 terminal.go:GOOS() - 0s
[TRACE] 08:14:02.078 terminal.go:GOOS() - 0s
[TRACE] 08:14:02.078 terminal.go:GOOS() - 0s
[TRACE] 08:14:02.078 terminal.go:GOOS() - 0s
[TRACE] 08:14:02.078 terminal.go:PathSeparator() - 0s
[TRACE] 08:14:02.078 terminal.go:GOOS() - 0s
[TRACE] 08:14:02.078 terminal.go:GOOS() - 0s
[TRACE] 08:14:02.078 terminal.go:GOOS() - 0s
[TRACE] 08:14:02.078 terminal.go:GOOS() - 0s
[TRACE] 08:14:02.078 terminal.go:GOOS() - 0s
[TRACE] 08:14:02.078 terminal.go:GOOS() - 0s
[TRACE] 08:14:02.078 terminal.go:GOOS() - 0s
[TRACE] 08:14:02.078 terminal.go:Flags() - 0s
[TRACE] 08:14:02.078 terminal.go:TemplateCache() - 0s
[TRACE] 08:14:02.078 text.go:Render({{ if .WSL }}WSL at {{ end }}{{.Icon}} ) - 1.6208ms
[TRACE] 08:14:02.078 text.go:Render({{ if .Root }}red{{ end }}) - 0s
[DEBUG] 08:14:02.078 path.go:colorizePath:728 → cycle: []
[DEBUG] 08:14:02.078 path.go:getFolderSeparator:263 → NO DATA
[DEBUG] 08:14:02.078 path.go:getFolderSeparator:265 →  
[DEBUG] 08:14:02.078 path.go:colorizePath:731 → cycle_folder_separator: false
[DEBUG] 08:14:02.078 path.go:colorizePath:732 → %s
[DEBUG] 08:14:02.078 path.go:colorizePath:734 → %s
[DEBUG] 08:14:02.078 path.go:colorizePath:735 → %s
[DEBUG] 08:14:02.078 path.go:colorizePath:736 → %s
[TRACE] 08:14:02.078 terminal.go:TemplateCache() - 0s
[TRACE] 08:14:02.078 terminal.go:GOOS() - 0s
[TRACE] 08:14:02.078 terminal.go:StackCount() - 0s
[TRACE] 08:14:02.078 text.go:Render({{ if not .Segments.Path.Writable }}white{{ end }}) - 694.6µs
[TRACE] 08:14:02.078 text.go:Render(diamond) - 0s
[TRACE] 08:14:02.078 text.go:Render({{ if .Root }}red{{ end }}) - 0s
[TRACE] 08:14:02.079 text.go:Render({{ if not .Segments.Path.Writable }}white{{ end }}) - 507.8µs
[DEBUG] 08:14:02.079 win32_windows.go:isWriteable:258 → not current user or in group
[DEBUG] 08:14:02.079 win32_windows.go:isWriteable:262 → current user is member of S-1-5-32-544
[DEBUG] 08:14:02.079 win32_windows.go:isWriteable:270 ↓
    WRITE_DAC
    WRITE_OWNER
    SYNCHRONIZE
    DELETE
    READ_CONTROL
[DEBUG] 08:14:02.079 win32_windows.go:isWriteable:272 → user has write access
[TRACE] 08:14:02.079 terminal_windows.go:DirIsWritable() - 1.3476ms
[TRACE] 08:14:02.079 terminal.go:TemplateCache() - 0s
[TRACE] 08:14:02.080 terminal.go:TemplateCache() - 0s
[TRACE] 08:14:02.080 text.go:Render( {{ path .Path .Location }} ) - 451.2µs
[TRACE] 08:14:02.080 text.go:Render(diamond) - 0s
[DEBUG] 08:14:02.135 terminal.go:CommandPath:395 → C:\Program Files\Git\cmd\git.exe
[TRACE] 08:14:02.135 terminal.go:CommandPath(git.exe) - 58.6701ms
[TRACE] 08:14:02.135 terminal.go:HasCommand(git.exe) - 58.6701ms
[DEBUG] 08:14:02.135 git.go:shouldDisplay:326 → fetch_bare_info: false
[DEBUG] 08:14:02.135 terminal.go:ResolveSymlink:287 → C:\Users\stanio
[TRACE] 08:14:02.135 terminal.go:ResolveSymlink(C:\Users\stanio) - 0s
[ERROR] 08:14:02.136 terminal.go:HasParentFilePath:559 → CreateFile .git: The system cannot find the file specified.
[TRACE] 08:14:02.136 terminal.go:HasParentFilePath(.git) - 1.0004ms
[TRACE] 08:14:02.136 text.go:Render(diamond) - 0s
[TRACE] 08:14:02.136 terminal.go:Shell() - 0s
[TRACE] 08:14:02.136 terminal.go:Shell() - 0s
[TRACE] 08:14:02.136 terminal.go:Flags() - 0s
[DEBUG] 08:14:02.136 terminal_windows.go:56 → C:\Users\stanio
[TRACE] 08:14:02.136 terminal.go:GOOS() - 0s
[DEBUG] 08:14:02.136 segment.go:Execute:103 → segment: Text
[TRACE] 08:14:02.136 terminal.go:TemplateCache() - 0s
[TRACE] 08:14:02.136 text.go:Render({{ if eq .Shell "cmd" }}❯{{ else if .Root }}#{{ else }}${{ end }} ) - 0s
[TRACE] 08:14:02.136 text.go:Render(plain) - 0s
[TRACE] 08:14:02.136 terminal.go:Shell() - 0s
[TRACE] 08:14:02.136 terminal.go:Shell() - 0s
[TRACE] 08:14:02.136 terminal.go:Shell() - 0s
[DEBUG] 08:14:02.136 terminal.go:Getenv:161 → C:\Users\stanio\AppData\Local
[TRACE] 08:14:02.136 terminal.go:Getenv(LOCALAPPDATA) - 0s
[TRACE] 08:14:02.136 terminal.go:CachePath() - 0s
[TRACE] 08:14:02.136 terminal.go:Flags() - 0s
```

--- a/src/cli/init.go
+++ b/src/cli/init.go
@@ -51,43 +51,7 @@ See the documentation to initialize your shell: https://ohmyposh.dev/docs/instal
 				return
 			}
 
-			var startTime time.Time
-			if debug {
-				startTime = time.Now()
-			}
-
-			env := &runtime.Terminal{
-				CmdFlags: &runtime.Flags{
-					Shell:  args[0],
-					Config: configFlag,
-					Strict: strict,
-					Debug:  debug,
-				},
-			}
-
-			env.Init()
-			defer env.Close()
-
-			template.Init(env)
-
-			cfg := config.Load(env)
-
-			feats := cfg.Features()
-
-			var output string
-
-			switch {
-			case printOutput, debug:
-				output = shell.PrintInit(env, feats, &startTime)
-			default:
-				output = shell.Init(env, feats)
-			}
-
-			if silent {
-				return
-			}
-
-			fmt.Print(output)
+			runInit(args[0])
 		},
 	}
 
@@ -98,4 +62,44 @@ See the documentation to initialize your shell: https://ohmyposh.dev/docs/instal
 	_ = initCmd.MarkPersistentFlagRequired("config")
 
 	return initCmd
+}
+
+func runInit(sh string) {
+	var startTime time.Time
+	if debug {
+		startTime = time.Now()
+	}
+
+	env := &runtime.Terminal{
+		CmdFlags: &runtime.Flags{
+			Shell:  sh,
+			Config: configFlag,
+			Strict: strict,
+			Debug:  debug,
+		},
+	}
+
+	env.Init()
+	defer env.Close()
+
+	template.Init(env)
+
+	cfg := config.Load(env)
+
+	feats := cfg.Features()
+
+	var output string
+
+	switch {
+	case printOutput, debug:
+		output = shell.PrintInit(env, feats, &startTime)
+	default:
+		output = shell.Init(env, feats)
+	}
+
+	if silent {
+		return
+	}
+
+	fmt.Print(output)
 }

--- a/src/cli/root.go
+++ b/src/cli/root.go
@@ -2,6 +2,7 @@ package cli
 
 import (
 	"os"
+	"strings"
 
 	"github.com/spf13/cobra"
 )
@@ -10,6 +11,8 @@ var (
 	configFlag string
 	shellName  string
 	silent     bool
+	// Deprecated flags, should be kept to avoid breaking CLI integration.
+	initialize bool
 )
 
 var RootCmd = &cobra.Command{
@@ -20,6 +23,11 @@ It can use the same configuration everywhere to offer a consistent
 experience, regardless of where you are. For a detailed guide
 on getting started, have a look at the docs at https://ohmyposh.dev`,
 	Run: func(cmd *cobra.Command, _ []string) {
+		if initialize {
+			runInit(strings.ToLower(shellName))
+			return
+		}
+
 		_ = cmd.Help()
 	},
 }
@@ -33,6 +41,10 @@ func Execute() {
 
 func init() {
 	RootCmd.PersistentFlags().StringVarP(&configFlag, "config", "c", "", "config file path")
+
+	// Deprecated flags, should be kept to avoid breaking CLI integration.
+	RootCmd.Flags().BoolVarP(&initialize, "init", "i", false, "init")
+	RootCmd.Flags().StringVarP(&shellName, "shell", "s", "", "shell")
 
 	// Hide flags that are deprecated or for internal use only.
 	RootCmd.PersistentFlags().BoolVar(&silent, "silent", false, "do not print anything")

--- a/src/config/segment.go
+++ b/src/config/segment.go
@@ -3,12 +3,14 @@ package config
 import (
 	"encoding/json"
 	"fmt"
+	"slices"
 	"strings"
 	"time"
 
 	"github.com/jandedobbeleer/oh-my-posh/src/cache"
 	"github.com/jandedobbeleer/oh-my-posh/src/color"
 	"github.com/jandedobbeleer/oh-my-posh/src/properties"
+	"github.com/jandedobbeleer/oh-my-posh/src/regex"
 	"github.com/jandedobbeleer/oh-my-posh/src/runtime"
 	"github.com/jandedobbeleer/oh-my-posh/src/template"
 
@@ -40,7 +42,7 @@ type Segment struct {
 	env                    runtime.Environment
 	Properties             properties.Map `json:"properties,omitempty" toml:"properties,omitempty"`
 	Cache                  *cache.Config  `json:"cache,omitempty" toml:"cache,omitempty"`
-	Style                  SegmentStyle   `json:"style,omitempty" toml:"style,omitempty"`
+	Alias                  string         `json:"alias,omitempty" toml:"alias,omitempty"`
 	styleCache             SegmentStyle
 	name                   string
 	LeadingDiamond         string         `json:"leading_diamond,omitempty" toml:"leading_diamond,omitempty"`
@@ -52,18 +54,19 @@ type Segment struct {
 	Background             color.Ansi     `json:"background,omitempty" toml:"background,omitempty"`
 	Filler                 string         `json:"filler,omitempty" toml:"filler,omitempty"`
 	Type                   SegmentType    `json:"type,omitempty" toml:"type,omitempty"`
-	Alias                  string         `json:"alias,omitempty" toml:"alias,omitempty"`
+	Style                  SegmentStyle   `json:"style,omitempty" toml:"style,omitempty"`
 	LeadingPowerlineSymbol string         `json:"leading_powerline_symbol,omitempty" toml:"leading_powerline_symbol,omitempty"`
-	ForegroundTemplates    template.List  `json:"foreground_templates,omitempty" toml:"foreground_templates,omitempty"`
 	Tips                   []string       `json:"tips,omitempty" toml:"tips,omitempty"`
+	ForegroundTemplates    template.List  `json:"foreground_templates,omitempty" toml:"foreground_templates,omitempty"`
 	BackgroundTemplates    template.List  `json:"background_templates,omitempty" toml:"background_templates,omitempty"`
 	Templates              template.List  `json:"templates,omitempty" toml:"templates,omitempty"`
 	ExcludeFolders         []string       `json:"exclude_folders,omitempty" toml:"exclude_folders,omitempty"`
 	IncludeFolders         []string       `json:"include_folders,omitempty" toml:"include_folders,omitempty"`
-	Duration               time.Duration  `json:"-" toml:"-"`
+	Needs                  []string       `json:"-" toml:"-"`
 	NameLength             int            `json:"-" toml:"-"`
 	MaxWidth               int            `json:"max_width,omitempty" toml:"max_width,omitempty"`
 	MinWidth               int            `json:"min_width,omitempty" toml:"min_width,omitempty"`
+	Duration               time.Duration  `json:"-" toml:"-"`
 	Interactive            bool           `json:"interactive,omitempty" toml:"interactive,omitempty"`
 	Enabled                bool           `json:"-" toml:"-"`
 	Newline                bool           `json:"newline,omitempty" toml:"newline,omitempty"`
@@ -94,6 +97,8 @@ func (segment *Segment) Execute(env runtime.Environment) {
 			segment.Duration = time.Since(start)
 		}()
 	}
+
+	defer segment.evaluateNeeds()
 
 	err := segment.MapSegmentWithWriter(env)
 	if err != nil || !segment.shouldIncludeFolder() {
@@ -310,4 +315,31 @@ func (segment *Segment) cwdIncluded() bool {
 
 func (segment *Segment) cwdExcluded() bool {
 	return segment.env.DirMatchesOneOf(segment.env.Pwd(), segment.ExcludeFolders)
+}
+
+func (segment *Segment) evaluateNeeds() {
+	value := segment.Template
+
+	if len(segment.ForegroundTemplates) != 0 {
+		value += strings.Join(segment.ForegroundTemplates, "")
+	}
+
+	if len(segment.BackgroundTemplates) != 0 {
+		value += strings.Join(segment.BackgroundTemplates, "")
+	}
+
+	if !strings.Contains(value, ".Segments.") {
+		return
+	}
+
+	matches := regex.FindAllNamedRegexMatch(`\.Segments\.(?P<NAME>[a-zA-Z0-9]+)`, value)
+	for _, name := range matches {
+		segmentName := name["NAME"]
+
+		if len(name) == 0 || slices.Contains(segment.Needs, segmentName) {
+			continue
+		}
+
+		segment.Needs = append(segment.Needs, segmentName)
+	}
 }

--- a/src/config/segment_test.go
+++ b/src/config/segment_test.go
@@ -160,3 +160,46 @@ func TestGetColors(t *testing.T) {
 		assert.Equal(t, tc.Expected, fgColor, tc.Case)
 	}
 }
+
+func TestEvaluateNeeds(t *testing.T) {
+	cases := []struct {
+		Segment *Segment
+		Case    string
+		Needs   []string
+	}{
+		{
+			Case: "No needs",
+			Segment: &Segment{
+				Template: "foo",
+			},
+		},
+		{
+			Case: "Template needs",
+			Segment: &Segment{
+				Template: "{{ .Segments.Git.URL }}",
+			},
+			Needs: []string{"Git"},
+		},
+		{
+			Case: "Template & Foreground needs",
+			Segment: &Segment{
+				Template:            "{{ .Segments.Git.URL }}",
+				ForegroundTemplates: []string{"foo", "{{ .Segments.Os.Icon }}"},
+			},
+			Needs: []string{"Git", "Os"},
+		},
+		{
+			Case: "Template & Foreground & Background needs",
+			Segment: &Segment{
+				Template:            "{{ .Segments.Git.URL }}",
+				ForegroundTemplates: []string{"foo", "{{ .Segments.Os.Icon }}"},
+				BackgroundTemplates: []string{"bar", "{{ .Segments.Exit.Icon }}"},
+			},
+			Needs: []string{"Git", "Os", "Exit"},
+		},
+	}
+	for _, tc := range cases {
+		tc.Segment.evaluateNeeds()
+		assert.Equal(t, tc.Needs, tc.Segment.Needs, tc.Case)
+	}
+}

--- a/src/prompt/segments.go
+++ b/src/prompt/segments.go
@@ -3,10 +3,8 @@ package prompt
 import (
 	"runtime"
 	"slices"
-	"strings"
 
 	"github.com/jandedobbeleer/oh-my-posh/src/config"
-	"github.com/jandedobbeleer/oh-my-posh/src/regex"
 	"github.com/jandedobbeleer/oh-my-posh/src/terminal"
 )
 
@@ -113,12 +111,7 @@ func (e *Engine) writeSegment(index int, block *config.Block, segment *config.Se
 }
 
 func (e *Engine) canRenderSegment(segment *config.Segment, executed []string) bool {
-	if !strings.Contains(segment.Template, ".Segments.") {
-		return true
-	}
-
-	matches := regex.FindNamedRegexMatch(`\.Segments\.(?P<NAME>[a-zA-Z0-9]+)`, segment.Template)
-	for _, name := range matches {
+	for _, name := range segment.Needs {
 		if slices.Contains(executed, name) {
 			continue
 		}

--- a/src/prompt/segments_test.go
+++ b/src/prompt/segments_test.go
@@ -35,38 +35,35 @@ func TestRenderBlock(t *testing.T) {
 
 func TestCanRenderSegment(t *testing.T) {
 	cases := []struct {
-		Case             string
-		Template         string
-		ExecutedSegments []string
-		Expected         bool
+		Case     string
+		Executed []string
+		Needs    []string
+		Expected bool
 	}{
 		{
 			Case:     "No cross segment dependencies",
 			Expected: true,
-			Template: "Hello",
 		},
 		{
 			Case:     "Cross segment dependencies, nothing executed",
 			Expected: false,
-			Template: "Hello {{ .Segments.Foo.World }} {{ .Segments.Foo.Bar }}",
+			Needs:    []string{"Foo"},
 		},
 		{
 			Case:     "Cross segment dependencies, available",
 			Expected: true,
-			Template: "Hello {{ .Segments.Foo.World }}",
-			ExecutedSegments: []string{
-				"Foo",
-			},
+			Executed: []string{"Foo"},
+			Needs:    []string{"Foo"},
 		},
 	}
 	for _, c := range cases {
 		segment := &config.Segment{
-			Type:     "text",
-			Template: c.Template,
+			Type:  "text",
+			Needs: c.Needs,
 		}
 
 		engine := &Engine{}
-		got := engine.canRenderSegment(segment, c.ExecutedSegments)
+		got := engine.canRenderSegment(segment, c.Executed)
 
 		assert.Equal(t, c.Expected, got, c.Case)
 	}

